openmc/mgxs_library.py

```diff
@@ -1634,6 +1685,267 @@ def set_inverse_velocity_mgxs(self, inverse_velocity, temperature=294.,
+        xsdata._xs_shapes = None
+
+        for i, temp in enumerate(xsdata.temperatures):
+            for xs in ['total', 'absorption', 'fission', 'nu_fission',
+                       'scatter_matrix', 'multiplicity_matrix',
+                       'prompt_nu_fission', 'delayed_nu_fission',
+                       'kappa_fission', 'chi', 'chi_prompt', 'chi_delayed',
+                       'beta', 'decay_rate', 'inverse_velocity']:
+                # Get the original data
+                orig_data = getattr(self, '_' + xs)[i]
+                if orig_data is not None:
+                    if target_representation == 'isotropic':
+                        # Since we are going from angle to isotropic, the
+                        # current data is just the average over the angle bins
+                        new_data = orig_data.mean(axis=(0, 1))
+                    elif target_representation == 'angle':
+                        # Since we are going from isotropic to angle, the
+                        # current data is just copied for every angle bin
+                        new_shape = (num_polar, num_azimuthal) + \
+                            orig_data.shape
+                        new_data = np.resize(orig_data, new_shape)
+                    setter = getattr(xsdata, 'set_' + xs)
+                    setter(new_data, temp)
+
+        return xsdata
+
+    def convert_scatter_format(self, target_format, target_order=None):
+        """Produce a new MGXSLibrary object with the same data, but converted
+        to the new scatter format and order
+
+        Parameters
+        ----------
+        target_format : {'tabular', 'legendre', 'histogram'}
+            Representation of the scattering angle distribution
+        target_order : int
+            Either the Legendre target_order, number of bins, or number of
+            points used to describe the angular distribution associated with
+            each group-to-group transfer probability
+
+        Returns
+        -------
+        openmc.XSdata
+            Multi-group cross section data with the same data as in self, but
+            represented as specified in :param:`target_format`.
+
+        """
+
+        from scipy.interpolate import interp1d
+        from scipy.integrate import simps
+        from scipy.special import eval_legendre
+
+        check_value('target_format', target_format, _SCATTER_TYPES)
+        check_type('target_order', target_order, Integral)
+        if target_format == 'legendre':
+            check_greater_than('target_order', target_order, 0, equality=True)
+        else:
+            check_greater_than('target_order', target_order, 0)
+
+        xsdata = copy.deepcopy(self)
+        xsdata.scatter_format = target_format
+        xsdata.order = target_order
+
+        # Reset and re-generate XSdata.xs_shapes with the new scattering format
+        xsdata._xs_shapes = None
+        xsdata.xs_shapes
+
+        for i, temp in enumerate(xsdata.temperatures):
```

Comment:
Ok vectorization isn't necessary for this PR of course. But my NumPy-ify everything OCD nature thinks it might be worth considering a single NumPy array for all temperatures in the future :-)